### PR TITLE
[MultiStatus] Improve error handling and two new methods to MultiStatus Class

### PR DESCRIPTION
**Pull request by sayan-das-in**
This PR adds two new methods to MultiStatusResponse class

- Adds two new methods to MultiStatusResponse class
- Adds error handling to handle an edge case when elements at an index could be undefined resulting in an exception

## Testing

Tested locally and with unit tests.

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/core/src/destination-kit/action.ts [hunk 1]
```diff
@@ -505,6 +505,18 @@ export class Action<Settings, Payload extends JSONLikeObject, AudienceSettings =
           }
 
           const response = performBatchResponse.getResponseAtIndex(resultsReadIndex++)
+          // We assume the response to be a failed response if it is undefined
+          // This is likely due to incorrect implementation of the MultiStatusResponse
+          if (!response) {
+            multiStatusResponse[i] = {
+              status: 500,
+              errormessage: 'MultiStatusResponse is missing a response at the specified index',
+              errortype: 'PAYLOAD_VALIDATION_FAILED',
```
Comment (varadarajan-tw) [inline packages/core/src/destination-kit/action.ts:514]: Could we use the enum for corresponding error code here?

**Review by varadarajan-tw — COMMENTED** [review]
LGTM! Left one comment.

File: packages/core/src/destination-kit/action.ts (hunk 1 above)
Comment (sayan-das-in) [reply packages/core/src/destination-kit/action.ts]: Updated this

Comment (sayan-das-in) [reply packages/core/src/destination-kit/action.ts]: Updated string literal to enum

**Review by varadarajan-tw — DISMISSED** [review]

**Review by varadarajan-tw — APPROVED** [review]

**Review by pooyaj — APPROVED** [review]
